main/app.py

```diff
@@ -0,0 +1,231 @@
+from __future__ import print_function
+import random
+import sqlite3
+import sys
+import tkFileDialog
+from Tkinter import Tk
+
+from clint.textui import colored, puts
+from colorama import init, Back, Style  # Fore
+from pyfiglet import figlet_format
+from termcolor import cprint
+from .models.person import Person, Fellow, Staff
+from .models.room import Office, LivingSpace
+import ipdb
+
+
+class Amity(object):
+    """Description for amity"""
+
+    def __init__(self, db_name="amity.sqlite"):
+
+        self.conn = sqlite3.connect(db_name)
+        self.connect = self.conn.cursor()
+        self.rooms = {
+            'O': {}, 'L': {}
+        }
+        self.people = {
+            'staff and fellow': [],
+            'fellow to house': []
+        }
+        # People who have not been allocated to an office or living space
+        self.unallocated = {
+            'O': [], 'L': []
+        }
+
+        # People that have been fully occupied
+        self.allocated_office = {}
+        self.allocated_living = {}
+
+        self.space = {}
+
+    def create_rooms(self, rooms, room_type=None):
+        """Allows user to enter a list of room names specifying
+                whether office or living spaces"""
+        if room_type is None:
+
+            room_type = raw_input(
+                "Enter room type: \n O: Office space \n L: Living space: \n"
+            )
+            while room_type != "O" and room_type != "L":
+                room_type = raw_input(
+                    "Try again. Enter Room Type:\n O: Office space \n L: Living space: \n"
+                )
+        random.shuffle(rooms)
+        for room_name in rooms:
+            if room_type.upper() == 'O':
+                self.space = Office(room_name)
+                self.rooms['O'].update({room_name: []})
+
+            else:
+                self.space = LivingSpace(room_name)
+                self.rooms['L'].update({room_name: []})
+        print(self.rooms)
+        print('New rooms succesfully created')
+
+    def add_person(self, first_name, last_name, person_type, want_housing):
+        name = first_name + " " + last_name
+        person_type = person_type
+        want_housing = want_housing
+        person = Person.create_person(name, person_type, want_housing)
+        if want_housing.upper() == 'Y':
+            self.people['fellow to house'].append(person)
+            self.allocate_room(person, 'L')
+        else:
+            self.people['staff and fellow'].append(person)
+        self.allocate_room(person, 'O')
+
+    def allocate_room(self, person, room_type):
+        """
+        Get the number of occupants,Alllocate the room till they are full, if room
+        filled remove it else alllocate room, then store unallocated"""
+        for room_name, people_space in self.rooms[room_type].iteritems():
+            check_capacity = len(people_space)
+            if not self.space.is_filled(check_capacity):
```

Comment:
**Code Quality: Nitpick**

What is happening here? I thought we already talked about making `is_filled()` a method of the room object?